File: javascript/pig-latin/pig-latin.js
export const translate = phrase => phrase.split(' ').map(word => word
  .replace(/^([^aeiou]*qu)(.*)/, '$2$1')
  .replace(/^(?!(?:xr|yt))([^aeiou]+)(.*)/, '$2$1')
  .replace(/^([^aeiou]+)(y.*)/, '$2$1') + 'ay'
).join(' ');

File: python/pig-latin/pig_latin.py
import re


def translate(text: str) -> str:
    words = text.split()
    for i, word in enumerate(words):
        if m := re.match(r"[aeiou]|xr|yt", word):
            words[i] += "ay"
        elif m := re.match(r"[^aeiou]*qu", word):
            words[i] = word[m.end(0) :] + m[0] + "ay"
        elif m := re.match(r"([^aeiou]+)y", word):
            words[i] = word[m.end(0) - 1 :] + m[0][:-1] + "ay"
        elif m := re.match(r"[^aeiou]+", word):
            words[i] = word[m.end(0) :] + m[0] + "ay"
    return " ".join(words)
